### PR TITLE
(towards #1779) Generalise NemoArrayAccess2Loop transformation to ArrayAcces2Loop

### PR DESCRIPTION
**Pull request by sergisiso**

### CODE REVIEW

**Comment by sergisiso** [comment]
@arporter This is ready for a first review, it continues generalising the remaining NEMO transformations. ArrayAccess2Loop was mostly used in the SIR transformation scripts. I have not run the generated SIR code, but nemo/eg4 can be used to diff the SIR output before and after this changes and see that only the variable names change with this PR.

File: src/psyclone/tests/psyir/transformations/arrayaccess2loop_trans_test.py [hunk 1]
```diff
@@ -187,17 +182,17 @@ def test_apply_third_dim_expr(tmpdir):
...
+        "    a(1,n,idx) = b(1,n,idx)\n"
         "  enddo\n\n")
     check_transformation(tmpdir, code, expected_result, index=2)
 
 
 def test_apply_fifth_dim_expr(tmpdir):
     '''Check that the expected code is produced for a 5D array where there
-    is no loop index information in the nemo api for the outermost
-    dimension so a new loop variable name is required.
+    is no loop index information for the outermost dimension so a new loo
```
Comment (arporter) [inline src/psyclone/tests/psyir/transformations/arrayaccess2loop_trans_test.py:194]: "loop"

Comment (arporter) [reply src/psyclone/tests/psyir/transformations/arrayaccess2loop_trans_test.py:194]: In fact, now that we're not looking up the name of the loop variable to use I think this test can be removed entirely?

**Review by arporter — CHANGES_REQUESTED** [review]
Good stuff Sergi, it's a nice feeling to see things getting shorter instead of longer :-)
Reference and User guides build fine.
I've set the integration tests going.
Just a few minor things to tidy up.

File: src/psyclone/psyir/transformations/allarrayaccess2loop_trans.py [hunk 2]
```diff
@@ -41,20 +41,20 @@
...
 from psyclone.psyir.nodes import Assignment
 from psyclone.psyir.transformations.transformation_error \
     import TransformationError
 
 
-class NemoAllArrayAccess2LoopTrans(Transformation):
+class AllArrayAccess2LoopTrans(Transformation):
     '''Provides a transformation from a PSyIR Assignment containing
-    constant index accesses to an array into single trip loops: For
+    constant index accesses to an array into single trip loops. For
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply src/psyclone/psyir/transformations/allarrayaccess2loop_trans.py:54]: Done

File: src/psyclone/psyir/transformations/allarrayaccess2loop_trans.py [hunk 3]
```diff
@@ -41,20 +41,20 @@
 
 from __future__ import absolute_import
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply src/psyclone/psyir/transformations/allarrayaccess2loop_trans.py:42]: removed

File: src/psyclone/psyir/transformations/allarrayaccess2loop_trans.py [hunk 4]
```diff
@@ -118,29 +116,27 @@ def apply(self, node, options=None):
...
         if not isinstance(node, Assignment):
             raise TransformationError(
-                f"Error in NemoAllArrayAccess2LoopTrans transformation. The "
+                f"Error in AllArrayAccess2LoopTrans transformation. The "
                 f"supplied node argument should be a PSyIR Assignment, "
                 f"but found '{type(node).__name__}'.")
 
     def __str__(self):
         return (
             "Convert the constant indices of a PSyIR array assignment into "
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply src/psyclone/psyir/transformations/allarrayaccess2loop_trans.py:138]: Done

File: src/psyclone/tests/psyir/transformations/arrayaccess2loop_trans_test.py (hunk 1 above)
Comment (sergisiso) [reply src/psyclone/tests/psyir/transformations/arrayaccess2loop_trans_test.py:194]: Agreed, removed the test.

**Comment by sergisiso** [comment]
@arporter This is ready for another review. I also removed a couple of configuration utility functions that Codecov said were not used anymore (and was planning to remove in an upcoming PR anyway)

**Review by arporter — APPROVED** [review]
All requested changes have been made, thanks.
Integration tests were fine.
Ref. guide still builds fine.
Will proceed to merge.

**Pull request merged**